### PR TITLE
Feature/init

### PR DESCRIPTION
**Pull request by ctrlaltd3v**

### CODE REVIEW

File: src/Altinn.Dan.Plugin.Statensvegvesen/Clients/Svv.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private static SvvResponse MapToInternal(KjoretoysokResponse externalResponse)
+    {
+        var internalResponse = new SvvResponse { Vehicles = new List<Vehicle>() };
+        if (externalResponse != null)
+            foreach (var kjoretoyResponse in externalResponse?.Kjoretoyresponser)
+            {
+                var technicalApproval = kjoretoyResponse.Kjoretoy?.Godkjenning?.TekniskGodkjenning;
+                internalResponse.Vehicles.Add(new Vehicle
+                {
+                    RegNr = kjoretoyResponse.Kjoretoy?.KjoretoyId?.Kjennemerke,
```
Comment (erlendoksvoll) [inline src/Altinn.Dan.Plugin.Statensvegvesen/Clients/Svv.cs]: RegNo is the English abbreviation

File: src/Altinn.Dan.Plugin.Statensvegvesen/Metadata.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                EvidenceSource = SOURCE,
+                BelongsToServiceContexts = new List<string> { "OED" },
+                RequiredScopes = "",
+                Values = new List<EvidenceValue>
+                {
+                    new()
+                    {
+                        EvidenceValueName = "default",
+                        ValueType = EvidenceValueType.JsonSchema,
+                        JsonSchemaDefintion = JsonConvert.SerializeObject(new SvvResponse())
```
Comment (erlendoksvoll) [inline src/Altinn.Dan.Plugin.Statensvegvesen/Metadata.cs]: Maybe something like JsonSchema.FromType<Type>().ToJson(Newtonsoft.Json.Formatting.Indented)

**Review by erlendoksvoll — CHANGES_REQUESTED** [review]

File: src/Altinn.Dan.Plugin.Statensvegvesen/Metadata.cs (hunk 2 above)
Comment (ctrlaltd3v) [reply src/Altinn.Dan.Plugin.Statensvegvesen/Metadata.cs]: Absolutely, it should be a SCHEMA.
Fix commited.

**Comment by ctrlaltd3v** [comment]
Take a look at the changes for handling of missing dates.

File: src/Altinn.Dan.Plugin.Statensvegvesen/Utils/Utils.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+public static class OedUtils
+{
+    public static string MapSsn(string ssn, string externalSource)
+    {
+        // todo Make this table driven and more maintainable when we have more interfaces and test users
+        // todo Differntiate between different test enviroments. Prod should allways return the input without modification.
+        if (externalSource == "svv")
+            switch (ssn)
+            {
+                case "12113701569":
```
Comment (erlendoksvoll) [inline src/Altinn.Dan.Plugin.Statensvegvesen/Utils/Utils.cs]: Could just remove the switch and keep only the svv values? Perhaps I should do the same for banking ... :|

Comment (ctrlaltd3v) [reply src/Altinn.Dan.Plugin.Statensvegvesen/Utils/Utils.cs]: Agree, looking into it :-)

Comment (ctrlaltd3v) [reply src/Altinn.Dan.Plugin.Statensvegvesen/Utils/Utils.cs]: Removed all of this :-D

**Comment by ctrlaltd3v** [comment]
How about this contraption? Homemade client, I guess it needs a bit of honing, but it's a lot smaller than the generated one. (Maybe easier to maintain..)

File: src/Altinn.Dan.Plugin.Statensvegvesen/Clients/SvvClient.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    case HttpStatusCode.OK:
+                    {
+                        var kjoretoysokResponse = JsonConvert.DeserializeObject<KjoretoysokResponse>(await response.Content.ReadAsStringAsync());
+
+                        return kjoretoysokResponse;
+                    }
+                    case HttpStatusCode.NoContent:
+                    {
+                        throw new EvidenceSourcePermanentClientException(Metadata.ERROR_NO_VEHICLES_FOUND, $"No vehicles found");
+                    }
```
Comment (erlendoksvoll) [inline src/Altinn.Dan.Plugin.Statensvegvesen/Clients/SvvClient.cs:46]: The api returns nocontent when no vehicles are found?

Comment (ctrlaltd3v) [reply src/Altinn.Dan.Plugin.Statensvegvesen/Clients/SvvClient.cs:46]: Yup.
From the spec:
"responses":
          "200": 
            "description": "Søk gjennomført, responsen inneholder eventuelle treff og/eller responskoder"
          "204": 
            "description": "Ingen kjøretøy funnet"
          "400": 
            "description": "Parameterfeil, responsen inneholder kun beskrivelse av feil som responskoder"

Comment (erlendoksvoll) [reply src/Altinn.Dan.Plugin.Statensvegvesen/Clients/SvvClient.cs:46]: Maybe we still should return an empty list instead of a permanentexception?

**Review by erlendoksvoll — APPROVED** [review]

**Pull request merged**